docs/io/ascii/fixed_width_gallery.rst

```diff
@@ -459,3 +459,105 @@ FixedWidthTwoLine
 
 ..
   EXAMPLE END
+
+Custom Header Rows
+==================
+
+The ``fixed_width`` and ``fixed_width_two_line`` formats normally include a
+single initial row with the column names in the header.  However, it is possible
+to customize the column attributes which appear as header rows. The available
+column attributes are ``name``, ``dtype``, ``format``, ``description`` and
+``unit``.  This is done by listing the desired the header rows using the
+``header_rows`` keyword argument.
+
+..
+  EXAMPLE START
+  Custom Header Rows with Fixed Width
+
+::
+    >>> from astropy.table.table_helpers import simple_table
+    >>> dat = simple_table(size=3, cols=4)
+    >>> dat["a"].info.unit = "m"
+    >>> dat["d"].info.unit = "m/s"
+    >>> dat["b"].info.format = ".2f"
+    >>> dat["c"].info.description = "C column"
+    >>> ascii.write(
+    ...    dat,
+    ...    format="fixed_width",
+    ...    header_rows=["dtype", "name", "unit", "format", "description"],
+    ... )
+    | int64 | float64 |      <U1 | int64 |
+    |     a |       b |        c |     d |
+    |     m |         |          | m / s |
+    |       |     .2f |          |       |
+    |       |         | C column |       |
+    |     1 |    1.00 |        c |     4 |
+    |     2 |    2.00 |        d |     5 |
+    |     3 |    3.00 |        e |     6 |
+
+In this example the 1st row is the ``dtype``, the 2nd row is the ``name``, and
+so forth. You must supply the ``name`` value in the ``header_rows`` list in
+order to get an output with the column name included.
+
+A table with non-standard header rows can be read back in the same way, using
+the same list of ``header_rows``::
+
+    >>> txt = """\
+    ... | int32 | float32 |      <U4 | uint8 |
+    ... |     a |       b |        c |     d |
+    ... |     m |         |          | m / s |
+    ... |       |     .2f |          |       |
+    ... |       |         | C column |       |
+    ... |     1 |    1.00 |        c |     4 |
+    ... |     2 |    2.00 |        d |     5 |
+    ... |     3 |    3.00 |        e |     6 |
+    ... """
+    >>> dat = ascii.read(
+    ...     txt,
+    ...     format="fixed_width",
+    ...     header_rows=["dtype", "name", "unit", "format", "description"],
+    ... )
+    >>> dat.info
+    <Table length=3>
+    name  dtype   unit format description
+    ---- ------- ----- ------ -----------
+    a   int32     m
+    b float32          .2f
+    c    str4                 C column
+    d   uint8 m / s
+
+..
+  EXAMPLE END
+
+..
+  EXAMPLE START
+  Custom Header Rows with Fixed Width Two Line
+
+The same idea can be used with the ``fixed_width_two_line`` format::
```

Comment:
```suggestion
The same idea can be used with the ``fixed_width_two_line`` format. While more than two lines 
in the header are expected when using the ``"header_rows"`` keyword, this format has its name from the fact
that the last line in the header (which is line two if no ``"header_rows"`` are given) marks the width of each column::
```

Reply:
I wanted to suggest an explanation of what "two_line" means in this context, because it's not too lines. Maybe my text suggestion is not the best, but I think something along those lines should go in there.

Reply:
Done.